e2e/scenario_win_test.go

```diff
@@ -112,53 +113,104 @@ func Test_Windows23H2Gen2(t *testing.T) {
-				ValidateCiliumIsNotRunningWindows(ctx, s)
+				ValidateFileHasContent(ctx, s, "/c/AzureData/CustomDataSetupScript.log", "CSEScriptsPackageUrl is https://packages.aks.azure.com/aks/windows/cse/aks-windows-cse-scripts-v0.0.52.zip")
 			},
 		},
 	})
 }
 
-func Test_Windows2025Gen2(t *testing.T) {
+func Test_Windows2022CachingRegression(t *testing.T) {
 	RunScenario(t, &Scenario{
-		Description: "Windows Server 2025 with Containerd - hyperv gen 2",
+		Description: "Windows 2022 VHD built before local cache enabled should still work - overwrite the CSE scripts package URL",
 		Config: Config{
-			Cluster:                ClusterAzureNetwork,
-			VHD:                    config.VHDWindows2025Gen2,
-			VMConfigMutator:        EmptyVMConfigMutator,
-			BootstrapConfigMutator: EmptyBootstrapConfigMutator,
+			Cluster:         ClusterAzureNetwork,
+			VHD:             config.VHDWindows2022ContainerdGen2,
+			VMConfigMutator: EmptyVMConfigMutator,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.CloudSpecConfig.KubernetesSpecConfig.CseScriptsPackageURL = "https://packages.aks.azure.com/aks/windows/cse/aks-windows-cse-scripts-v0.0.52.zip"
+			},
 			Validator: func(ctx context.Context, s *Scenario) {
-				ValidateWindowsVersionFromWindowsSettings(ctx, s, "2025-gen2")
-				ValidateWindowsProductName(ctx, s, "Windows Server 2025 Datacenter")
-				ValidateWindowsDisplayVersion(ctx, s, "24H2")
-				ValidateFileHasContent(ctx, s, "/k/kubeletstart.ps1", "--container-runtime=remote")
-				ValidateWindowsProcessHasCliArguments(ctx, s, "kubelet.exe", []string{"--rotate-certificates=true", "--client-ca-file=c:\\k\\ca.crt"})
-				ValidateCiliumIsNotRunningWindows(ctx, s)
+				ValidateFileHasContent(ctx, s, "/AzureData/CustomDataSetupScript.log", "CSEScriptsPackageUrl is https://packages.aks.azure.com/aks/windows/cse/aks-windows-cse-scripts-v0.0.52.zip")
+			},
+		},
+	})
+}
+
+func Test_Windows2019CachingRegression(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "Windows 2019 VHD built before local cache enabled should still work - overwrite the CSE scripts package URL",
+		Config: Config{
+			Cluster:         ClusterAzureNetwork,
+			VHD:             config.VHDWindows2019Containerd,
+			VMConfigMutator: EmptyVMConfigMutator,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.CloudSpecConfig.KubernetesSpecConfig.CseScriptsPackageURL = "https://packages.aks.azure.com/aks/windows/cse/aks-windows-cse-scripts-v0.0.52.zip"
+			},
+			Validator: func(ctx context.Context, s *Scenario) {
+				ValidateFileHasContent(ctx, s, "/AzureData/CustomDataSetupScript.log", "CSEScriptsPackageUrl is https://packages.aks.azure.com/aks/windows/cse/aks-windows-cse-scripts-v0.0.52.zip")
 			},
 		},
 	})
 }
 
-// TODO: enable this test once production AKS supports Cilium Windows
+// TODO: enable this test once containerd2 config is properly updated.
+// func Test_Windows2025(t *testing.T) {
```

Comment:
Rather than comment this out, can you add the "t.Skipf" back so it's logged that we skip it?